lib/ec2imgutils/ec2publishimg.py

```diff
@@ -156,6 +158,47 @@ def publish_images(self):
         images = self._get_images()
 
         for image in images:
+            if image['State'] != 'available':
+                waiter = self._connect().get_waiter('image_available')
```

Comment:
A waiter does no block, so somewhere we have to sleep or we'll just raise through the loop.

Reply:
I tested this with different values of 

                        WaiterConfig={
                            'Delay': 15,
                            'MaxAttempts': 40
                        }

and it did block as expected